### PR TITLE
Coverity 1545947: Resource leak in header_rewrite plugin

### PR DESCRIPTION
**Pull request by bryancall**

### CODE REVIEW

File: plugins/header_rewrite/header_rewrite.cc [hunk 1]
```diff
@@ -199,23 +199,22 @@ RulesConfig::parse_config(const std::string &fname, TSHttpHookID default_hook, c
     }
 
     // If we are at the beginning of a new condition, save away the previous rule (but only if it has operators).
-    if (p.is_cond() && add_rule(rule)) {
-      rule = nullptr;
+    if (p.is_cond() && add_rule(rule.get())) {
+      rule.release();
```
Comment (maskit) [inline plugins/header_rewrite/header_rewrite.cc:203]: It's unfortunate that we still have to call `release()`. I wondered if we can `std::move` the `rule` and call `release()` in `add_rule` but it looks like the following code relies on whether `rule` becomes nullptr.

**Review by maskit — APPROVED** [review]

**Comment by bryancall** [comment]
[approve ci centos]

File: plugins/header_rewrite/header_rewrite.cc (hunk 1 above)
Comment (bryancall) [reply plugins/header_rewrite/header_rewrite.cc:203]: I wanted to check to return value of `add_rule()` before I released the pointer.   `add_rule()` checks to see if the `rule` is not a nullptr and has an operator before assigning the pointer and returning true..

**Comment by cmcfarlen** [comment]
Cherry-picked to v10.0.x

**Pull request merged**